include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,330 @@
+ * @tparam Robot[optional] derived `kodlab::RobotBase` class
+ */
+template<class Robot = kodlab::RobotBase>
+class BehaviorManager {
+  static_assert(std::is_base_of<kodlab::RobotBase, Robot>::value,
+                "Robot must have base kodlab::RobotBase.");
+
+ private:
+  /**
+   * @brief Robot that behaviors run on
+   */
+  std::shared_ptr<Robot> robot_;
+
+  /**
+   * @brief Behavior command that kills robot, default -1
+   */
+  const int KILL_ROBOT_IDX = -1;
+
+  /**
+   * @brief Name of the kill behavior
+   */
+  const std::string KILL_BEHAVIOR_NAME = "KILL";
+
+  /**
+   * @brief Behavior command that turns robot off, always 0
+   */
+  const int DEFAULT_IDX = 0;
+
+  /**
+   * @brief Name of the default behavior
+   */
+  std::string default_name_;
+
+  /**
+   * @brief Behaviors available to the robot
+   * @note The first element of this vector is always `OFF_BEHAVIOR`.  Added
+   *       behaviors are appended to the end.
+   */
+  std::vector<std::unique_ptr<Behavior<Robot>>> behaviors_;
+
+  /**
+   * @brief Map from behavior name to index in `behaviors_`
+   */
+  std::unordered_map<std::string, int> name_to_idx_map_;
+
+  /**
+   * @brief Names of behaviors in `behaviors_`
+   */
+  std::vector<std::string> names_;
+
+  /**
+   * @brief Index in `behaviors_` of the selected behavior
+   */
+  unsigned int selected_idx = 0;
+
+  /**
+   * @brief Index in `behaviors_` of the incoming behavior while transitioning
+   *        behaviors
+   */
+  int next_idx_ = 0;
+
+  /**
+   * @brief Indicates whether a behavior switch is in progress
+   */
+  bool switching_behaviors_ = false;
+
+  /**
+   * @brief Constructs and initializes a behavior
+   * @tparam BehaviorType `kodlab::Behavior`-derived type
+   * @tparam ConstructorArgs constructor arguments
+   * @param args behavior constructor arguments
+   * @return unique pointer to the behavior if initialization was successful,
+   *         nullptr otherwise
+   */
+  template<class BehaviorType, typename... ConstructorArgs>
+  std::unique_ptr<BehaviorType> ConstructBehavior(ConstructorArgs &&... args) {
+    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
+                  "BehaviorType must be a `kodlab::Behavior`-derived type");
+    auto b = std::make_unique<BehaviorType>(args...);
+    if (b->Init()) {
+      b->set_initialized();
+      return std::move(b);
+    } else { return nullptr; }
+  }
+
+ public:
+  /**
+   * @brief Construct a behavior manager object
+   * @param robot pointer to robot that the behavior is executing on
+   */
+  BehaviorManager(std::shared_ptr<Robot> robot) : robot_(robot) {
+    default_name_ = "OFF";
+    AddBehavior<OffBehavior<Robot>>(robot_, default_name_);
+  }
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~BehaviorManager() = default;
+
+  /**
+   * @brief Set the current behavior
+   * @param next_idx index in `behaviors_`
+   */
+  virtual void SetBehavior(const int &next_idx) {
```

Comment:
Don't pass int by reference: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-conventional

Reply:
Fixed in e1e7ee036a97fdee4c755d57904d3f9dc82879e4